react/PlaceDetails.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react'
+import { Dropdown } from 'vtex.styleguide'
+
+const address: { [index: string]: any } = {
+  street: 'Av. Belgrano',
+  number: 2248,
+  complement: '',
+  postalCode: '2000',
+  city: 'Rosario',
+  state: 'Santa Fe',
+}
+
+const countryDescriptions: CountryDescription[] = [
+  {
+    name: 'ARG',
+    summary: [
+      [{ name: 'street' }, { delimiter: ' ', name: 'number' }],
+      [{ name: 'complement' }],
+      [{ name: 'postalCode' }],
+      [{ name: 'city' }, { delimiter: ', ', name: 'state' }],
+    ],
+  },
+  {
+    name: 'BRA',
+    summary: [
+      [
+        { name: 'street' },
+        { delimiter: ' ', name: 'number' },
+        { delimiter: ', ', name: 'complement' },
+      ],
+      [
+        { name: 'neighborhood', delimiterAfter: ' - ' },
+        { name: 'city' },
+        { delimiter: ' - ', name: 'state' },
+      ],
+      [{ name: 'postalCode' }],
+    ],
+  },
+  {
+    name: 'KOR',
+    summary: [
+      [{ name: 'street' }, { delimiter: ', ', name: 'complement' }],
+      [
+        { name: 'city' },
+        { delimiter: ', ', name: 'state' },
+        { delimiter: ' ', name: 'postalCode' },
+      ],
+    ],
+  },
+]
+
+const PlaceDetails: StorefrontFunctionComponent<PlaceDetailsProps> = () => {
+  const [option, setOption] = useState<string>('ARG')
+  const [summary, setSummary] = useState<LineComponent[][]>([])
+
+  return (
+    <div>
+      <Dropdown
+        label="Country"
+        options={[
+          { value: 'ARG', label: 'Argentina' },
+          { value: 'BRA', label: 'Brazil' },
+          { value: 'KOR', label: 'Korea' },
+        ]}
+        value={option}
+        onChange={(_: any, newVal: string) => {
+          setOption(newVal)
+          let newSummary = countryDescriptions.find(
+            description => description.name == newVal
+          )
+          setSummary(newSummary ? newSummary.summary : [])
+        }}
```

Comment:
I would move this to separate function to improve code readability.

Reply:
I moved it to a function outside, but still inside the PlaceDetails function component to be able to use the setOption and setSummary functions.